assets/app/scripts/controllers/projects.js

```diff
@@ -24,6 +24,45 @@ angular.module('openshiftConsole')
       $('#openshift-logo').off('click.projectsPage');
     });
 
+    $scope.toggleDelete = function(project) {
+      // to toggle prompt to delete
+      // disable the delete button if current project is being deleted
+      if (!$scope.projects[project]['deleting']) {
+        $scope.projects[project]['toggled'] = !$scope.projects[project]['toggled'];
+      }
+    };
+
+    $scope.deleteProject = function(project) {
+      $scope.alerts = {};
+      $scope.projects[project]['toggled'] = false;
+      $scope.projects[project]['deleting'] = true;
+      DataService.delete('projects', project, $scope)
+      .then(function() {
+        $scope.alerts = [{
+          type: "success",
+          message: "Project " + $scope.projects[project]['metadata']['annotations']['openshift.io/display-name'] + " was successfully deleted."
+        }];
+        delete $scope.projects[project];
+        $timeout(function() {
+          $scope.alerts = {};
+        }, 10000);
+      }, function(result) {
```

Comment:
@liggitt what would be the advantage of using `.catch()` over a 2nd arg?

i'm curious because the 2nd arg method is how it seems to be handled in most of the assets codebase, especially when it comes to sending alerts.

examples:

* [line 24, create project controller](https://github.com/openshift/origin/blob/e3fc774910e54898c182a825d2c238ce81e2859c/assets/app/scripts/controllers/createProject.js#L24)
* [line 127, builds controller](https://github.com/openshift/origin/blob/d78f443c547bdf2635b8c8d439326e3191ec5362/assets/app/scripts/controllers/builds.js#L127)
* [line 130, new from template controller](https://github.com/openshift/origin/blob/4beb4565322eeb13fc1cb13f637c8663f2842481/assets/app/scripts/controllers/newfromtemplate.js#L130)

and a few others.